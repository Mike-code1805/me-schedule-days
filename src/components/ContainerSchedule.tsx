import React from 'react';
import styles from '../styles/containerSchedule/containerSchedule.modules.css';

export interface Props {
  title?: string;
  backgroundColor?: string;
  idValue?: string;
  onClickId?: any;
}

export const ContainerSchedule = ({
  title = 'ProgramaciÃ³n I',
  backgroundColor = 'red',
  idValue,
  onClickId,
}: Props) => {
  return (
    <div
      onClick={() => onClickId(idValue)}
      className={styles["containerSchedule"]}
      style={{ background: backgroundColor }}
    >
      <h1 className={styles["containerSchedule_text"]}>{title}</h1>
    </div>
  );
};
