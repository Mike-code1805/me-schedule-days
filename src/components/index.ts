import { Schedule as ScheduleComponent } from './Schedule';
import { ContainerSchedule } from './ContainerSchedule';
import { ScheduleHOCProps } from '../interfaces/schedules.interfaces';
import { DayContainerSchedule } from './DayContainerSchedule';

export const Schedule: ScheduleHOCProps = Object.assign(ScheduleComponent, {
  Container: ContainerSchedule,
  DayContainer: DayContainerSchedule,
});
