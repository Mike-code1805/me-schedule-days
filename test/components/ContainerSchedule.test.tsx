import React from 'react';
import renderer from 'react-test-renderer';
import { Schedule } from '../../src/components/index';

describe('ContainerSchedule', () => {
  test('should render ContainerSchedule title component', () => {
    const wrapper = renderer.create(
      <Schedule.Container title="ContainerSchedule Title" />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('should render ContainerSchedule color component', () => {
    const wrapper = renderer.create(
      <Schedule.Container backgroundColor="black" />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('should render ContainerSchedule color and title component', () => {
    const wrapper = renderer.create(
      <Schedule.Container title="ContainerSchedule Title" backgroundColor="black" />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
