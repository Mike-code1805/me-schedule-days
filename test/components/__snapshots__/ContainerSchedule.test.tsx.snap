// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContainerSchedule should render ContainerSchedule color and title component 1`] = `
<div
  className="container"
  style={
    Object {
      "background": "black",
    }
  }
>
  <h1
    className="container_text"
  >
    ContainerSchedule Title
  </h1>
</div>
`;

exports[`ContainerSchedule should render ContainerSchedule color component 1`] = `
<div
  className="container"
  style={
    Object {
      "background": "black",
    }
  }
>
  <h1
    className="container_text"
  >
    Programaci√≥n I
  </h1>
</div>
`;

exports[`ContainerSchedule should render ContainerSchedule title component 1`] = `
<div
  className="container"
  style={
    Object {
      "background": "red",
    }
  }
>
  <h1
    className="container_text"
  >
    ContainerSchedule Title
  </h1>
</div>
`;
